src/extensions/askUserForParam.ts

```diff
@@ -116,6 +129,81 @@ export async function askForParam(paramSpec: Param): Promise<string> {
   return response;
 }
 
+async function promptSecret(
+  projectId: string,
+  instanceId: string,
+  paramSpec: Param
+): Promise<string> {
+  if (paramSpec.reconfiguring) {
+    const action = await promptOnce({
+      type: "list",
+      message: `Choose what you would like to do with this secret:`,
+      choices: [
+        { name: "Leave unchanged", value: SecretUpdateAction.LEAVE },
+        { name: "Set new value", value: SecretUpdateAction.SET_NEW },
+      ],
+    });
+    switch (action) {
+      case SecretUpdateAction.SET_NEW:
+        let secret;
+        let secretName;
+        if (paramSpec.default) {
+          secret = secretManagerApi.parseSecretResourceName(paramSpec.default);
+          secretName = secret.name;
+        } else {
+          secretName = await generateSecretName(projectId, instanceId, paramSpec.param);
+        }
+
+        const secretValue = await promptOnce({
+          name: paramSpec.param,
+          type: "input",
```

Comment:
promptOnce uses inquirer (https://www.npmjs.com/package/inquirer) under the hood, so it supports "password" as a type, which would obscure this text. Not sure if you want to use it here, but I figured its a UX  question worth considering since this is potentially sensitive.

Reply:
Good to know, but I don't think we want to hide the value during entry.